src/Altinn.App.Api/Controllers/ProcessController.cs

```diff
@@ -202,15 +203,15 @@ public class ProcessController : ControllerBase
             }
         }
 
-        private async Task<bool> CanTaskBeEnded(Instance instance, string currentElementId)
+        private async Task<bool> CanTaskBeEnded(Instance instance, string currentTaskId)
         {
             List<ValidationIssue> validationIssues = new List<ValidationIssue>();
 
             bool canEndTask;
 
             if (instance.Process?.CurrentTask?.Validated == null || !instance.Process.CurrentTask.Validated.CanCompleteTask)
             {
-                validationIssues = await _validationService.ValidateAndUpdateProcess(instance, currentElementId);
+                validationIssues = await _validationService.ValidateInstanceAtTask(instance, currentTaskId);
```

Comment:
I've stopped updating `instance.Process.CurrentTask.Validated.CanCompleteTask` in the `_validationSerivce` and not added it back in anywhere, so we should either add it back or remove this check. It really does not make any sense to cache `CanCompleteTask` without locking all data elements that the validation is dependent on.

Reply:
So now we run the actual validations when we try to move the process to the next step instead of relying on a possible outdated status?

Reply:
Yes, when I've stopped updating the status, we need to run validation again before processing next. We either need to remove the check here, or add back the caching when we validate.

Reply:
Yes, I'll remove this check. Question is what to do with instance.Process.CurrentTask.Validated.CanCompleteTask, if we don't use it anymore.

Reply:
I think we need to leave it, as the object/field is "owned" by storage and storage needs to support older versions of app-lib that rely on it.